extension/background/pageMetadata.js

```diff
@@ -14,3 +14,12 @@ export async function getMetadata(tabId) {
     type: "getMetadata",
   });
 }
+
+export async function cleanURL(url) {
+  await content.lazyInject(url, "/background/pageMetadata-contentScript.js");
+  browser.webRequest.onBeforeRequest.addListener(
+    cleanURL,
+    { urls: ["<all_urls>"] },
+    ["blocking"]
+  );
```

Comment:
We don't need or want to actually change the URLs in the browser for this, just to change the URL for the purposes of copying the link. So if you go to a page with `?utm_campaign=whatever` and then say "copy link", that utm_campaign won't be in the copied link.

So all this code can actually live in `pageMetadata.js` and instead of this:

```js
  return browser.tabs.sendMessage(tabId, {
    type: "getMetadata",
  });
```

it would be like:

```js
  const metadata = await browser.tabs.sendMessage(tabId, {
    type: "getMetadata",
  });
  metadata.canonical = cleanUrl(metadata.canonical);
  return metadata;
```